internal/server/server.go

```diff
@@ -21,6 +23,60 @@ import (
 
 type Handlers struct{}
 
+func OsbuildClient() (*cloudapi.Client, error) {
+	socket, ok := os.LookupEnv("OSBUILD_SERVICE")
+	if !ok {
+		socket = "http://127.0.0.1:80"
+	}
+	Log.Infoln("Creating osbuildclient with socket %s", socket)
+	return cloudapi.NewClient(socket, ConfigureClient)
+}
+
+func ConfigureClient(client *cloudapi.Client) error {
+	// set up client certificate authentication
+	if (strings.HasPrefix(client.Server, "https")) {
+		certFile, ok := os.LookupEnv("OSBUILD_CERT_PATH")
```

Comment:
This is a general comment, and I don't really know enough about the environment we expect, so take it with a grain of salt: in composer we try to shy away from environment variables and rather use config files. Is that something we could/should do here too, or is the env vars more standard / makes stuff easier in some way?

Either way, I think it is better if we extract the config from env vars or from config files in `main.go` and pass it down from there, rather than doing it somewhere deep down in `internal/` (admittedly not so deep at the moment, but once things grow it might get unwieldy).

Reply:
Since our main target is an openshift environment it's pretty flexible. Most services I've used as a reference use environment variables but there's nothing stopping us from using a file, as long as the variables are visible in openshift without having to open a shell, which we can easily do using a configmap which creates a file.

Reply:
Ok I tried addressing this in #41:

https://github.com/osbuild/image-builder/blob/e041ef90579f1d29cf48a371f0bc1e881d44e809/cmd/image-builder/config.go

I think using the environment will still be easiest for the openshift environment, but if we want to ever fill the config in from a file that won't be too hard then.

Reply:
Ah, just read about config maps and secrets, very cool. I think having config as files would be best as it is easier to extend, and env vars are just iffy in general. I wouldn't mind keeping secrets as vars as their structure is much more limited, and i think it is more standard? i think they could be made into files too though, if you wanted to

Reply:
Hmm I'd make a config.go and a secrets.go then i guess, the latter reading from the environment (which will be mapped by openshift from Vault), and then the former reading from a file.

Reply:
After our call (and also reading all your comments, I somehow only saw the first), I take back my suggestion regarding config files. Let's change that if and when it becomes relevant and keep it simple for now.